EDU-10878: FastStore v1 release note: fixed abandoned cart tracking (#754)
vtexdocs/dev-portal-content: docs/release-notes/2023-10-11-faststore-v1-abandoned-cart.md

## What needs to be done?

Stores using [FastStore v1](https://v1.faststore.dev/) must define the `VTEX_METADATA` object to ensure the correct behavior. To do so, open your FastStore project code and follow the steps below:

1. In the `src/sdk/analytics/index.tsx` file, add the highlighted code:

    ```tsx
    if (typeof window !== 'undefined') {
        window.dataLayer = window.dataLayer ?? []
        window.VTEX_METADATA = {
            account: storeConfig.api.storeId,
            renderer: 'faststore',
        }
    }
    ```

2. In the `src/typings/global.d.ts` file, add the highlighted code:

    ```ts
    declare module '*.scss';
    declare module '*.png';

    interface Window extends Window {
        dataLayer: any[];
        VTEX_METADATA: {
            account: string,
            renderer: "faststore"
        };
        sendrc: (eventName: string, eventValues?: any) => void
    }
    ```

3. In the terminal, run `yarn dev` to sync the changes.

> After syncing the changes, you can verify if the `VTEX_METADATA` has been defined and if the data captured by Return Customer (RC) is updating the records in Master Data.

Reviewer: ```suggestion
After syncing the changes, you can test the solution by checking the information in the following topics:
```

> - **Check the `VTEX_METADATA`**

Reviewer: ```suggestion
- **Check if `VTEX_METADATA` has been defined:**

```

  1. Access the provided localhost URL in your browser.

>   2. Open the Dev Tools terminal, go to the **Console** tab, and check if the `window.VTEX_METADATA` is defined in the console.

Reviewer: ```suggestion
  2. Open the Dev Tools terminal, go to the **Console** tab, and check if the `window.VTEX_METADATA` is defined in the console, as illustrated in the image below.
```

> ![console-object](https://vtexhelp.vtexassets.com/assets/docs/src/console-vtex-metadata___f1d9f78c17949decc87f22266ac3285d.png)

Author: ```suggestion
![console-object](https://vtexhelp.vtexassets.com/assets/docs/src/console-vtex-metadata___f1d9f78c17949decc87f22266ac3285d.gif)
```

> ![console-object](https://vtexhelp.vtexassets.com/assets/docs/src/console-vtex-metadata___f1d9f78c17949decc87f22266ac3285d.gif)

Reviewer: ```suggestion
  ![console-object](https://vtexhelp.vtexassets.com/assets/docs/src/console-vtex-metadata___f1d9f78c17949decc87f22266ac3285d.gif)
```

> - **Check if the data captured by Return Customer (RC) is updating the records in Master Data**
> 
> To check, refer to the topic [Is the data captured by RC updating the records in Master Data?](https://help.vtex.com/en/tutorial/configurando-carrinho-abandonado--tutorials_740?locale=pt#is-the-data-captured-by-rc-updating-the-records-in-master-data).

Reviewer: ```suggestion
- **Check if the data captured by Return Customer (RC) is updating the records in Master Data:**

   Refer to the topic [Is the data captured by RC updating the records in Master Data?](https://help.vtex.com/en/tutorial/setting-up-abandoned-carts--tutorials_740#is-the-data-captured-by-rc-updating-the-records-in-master-data).
```
